src/main/src/test/java/org/geoserver/catalog/ResourcePoolTest.java

```diff
@@ -890,4 +892,53 @@ public void testCoverageReaderInputConverter() throws IOException {
         // File object
         assertTrue(reader.getSource() instanceof File);
     }
+
+    /**
+     * Tests that even if the input string cannot be parsed as a valid URI reference, but the it is
+     * nontheless a valid file URL, the CoverageReaderFileConverter is able to do the conversion.
+     *
+     * @throws IOException
+     */
+    @Test
+    public void testCoverageReaderInputConverterInvalidURI() throws IOException {
+        Assume.assumeTrue(SystemUtils.IS_OS_WINDOWS);
+        String fileURL =
+                MockData.class.getResource("tazdem.tiff").getFile().replaceAll("/", "\\\\");
```

Comment:
Could be a little simpler: 

```suggestion
                MockData.class.getResource("tazdem.tiff").getFile().replace('/', '\\');
```

Reply:
fixed